tests/Interval/IntervalToStringTest.php

```diff
@@ -25,51 +25,102 @@ public function testZeroInterval()
         $this->assertEquals('PT0S', (string)$ci);
     }
 
+    public function testNegativeArguments()
+    {
+        $ci = new ChronosInterval(1, -2, 0, 15);
+        $this->assertEquals('P1Y15D', (string)$ci, 'Negative arguments are not considered');
+    }
+
+    /**
+     * Date section
+     */
     public function testYearInterval()
     {
         $ci = new ChronosInterval();
+        $ci1 = new ChronosInterval(1);
+        $ci2 = new ChronosInterval(0, 12, 0, 0);
+        $ci3 = new ChronosInterval(0, 0, 0, 365);
+        $ci4 = new ChronosInterval(0, 0, 52, 1);
+
         $this->assertEquals('P1Y', (string)$ci);
+        $this->assertEquals('P1Y', (string)$ci1);
+        $this->assertEquals('P1Y', (string)$ci2);
+        $this->assertEquals('P1Y', (string)$ci3);
+        $this->assertEquals('P1Y', (string)$ci4);
     }
 
     public function testMonthInterval()
     {
-        $ci = new ChronosInterval(0, 1);
-        $this->assertEquals('P1M', (string)$ci);
+        $ci1 = new ChronosInterval(0, 1);
+        $ci2 = new ChronosInterval(0, 0, 0, 30, 10);
+        $ci3 = new ChronosInterval(0, 0, 4, 2, 10);
+
+        $this->assertEquals('P1M', (string)$ci1);
+        $this->assertEquals('P1M', (string)$ci2);
+        $this->assertEquals('P1M', (string)$ci3);
     }
 
     public function testWeekInterval()
     {
-        $ci = new ChronosInterval(0, 0, 1);
-        $this->assertEquals('P7D', (string)$ci);
+        $ci1 = new ChronosInterval(0, 0, 1);
+        $ci2 = new ChronosInterval(0, 0, 0, 7);
+        $ci3 = new ChronosInterval(0, 0, 0, 0, 7 * 24);
+        $ci4 = new ChronosInterval(0, 0, 0, 0, 0, 7 * 24 * 60);
+
+        $this->assertEquals('P7D', (string)$ci1);
+        $this->assertEquals('P7D', (string)$ci2);
+        $this->assertEquals('P7D', (string)$ci3);
+        $this->assertEquals('P7D', (string)$ci4);
     }
 
     public function testDayInterval()
     {
-        $ci = new ChronosInterval(0, 0, 0, 1);
-        $this->assertEquals('P1D', (string)$ci);
+        $ci1 = new ChronosInterval(0, 0, 0, 1);
+        $ci2 = new ChronosInterval(0, 0, 0, 0, 24);
+        $ci3 = new ChronosInterval(0, 0, 0, 0, 0, 24 * 60);
+        $ci4 = new ChronosInterval(0, 0, 0, 0, 0, 0, 24 * 60 * 60);
+
+        $this->assertEquals('P1D', (string)$ci1);
+        $this->assertEquals('P1D', (string)$ci2);
+        $this->assertEquals('P1D', (string)$ci3);
+        $this->assertEquals('P1D', (string)$ci4);
     }
 
     public function testMixedDateInterval()
     {
-        $ci = new ChronosInterval(1, 2, 0, 3);
-        $this->assertEquals('P1Y2M3D', (string)$ci);
+        $ci1 = new ChronosInterval(1, 2, 0, 3);
+        $ci2 = new ChronosInterval(0, 14, 0, 3);
```

Comment:
Do you want to have a test where a smaller time measurement bleeds in to the next one? Such as `new ChronosInterval(1, 2, 0, 3, 36);`?